File: 11_1d_dynamic_programming/01_min_cost_climbing_staris/min_cost_climbing_staris.py
from typing import List


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        min_cost = [0] * (len(cost) + 1)
        i = 2
        while i < len(min_cost):
            min_cost[i] = min(cost[i - 1] + min_cost[i - 1],
                              cost[i - 2] + min_cost[i - 2])
            i += 1
        return min_cost[-1]

File: NeetCode/11_1d_dynamic_programming/01_min_cost_climbing_staris/min_cost_climbing_staris.cpp
#include <vector>

using namespace std;

class Solution
{
  public:
    int minCostClimbingStairs(vector<int>& cost)
    {
        vector<int> min_cost(cost.size() + 1, 0);
        for (int i = 2; i < min_cost.size(); ++i)
        {
            min_cost[i] = min(cost[i - 1] + min_cost[i - 1],
                              cost[i - 2] + min_cost[i - 2]);
        }
        return min_cost.back();
    }
};